Userland/Libraries/LibGUI/Notification.cpp

```diff
@@ -58,9 +58,56 @@ Notification::~Notification()
 
 void Notification::show()
 {
+    VERIFY(m_id == -1 && !m_showing);
+
     auto connection = NotificationServerConnection::construct();
     auto icon = m_icon ? m_icon->to_shareable_bitmap() : Gfx::ShareableBitmap();
-    connection->send_sync<Messages::NotificationServer::ShowNotification>(m_text, m_title, icon);
+    auto response = connection->send_sync<Messages::NotificationServer::ShowNotification>(m_text, m_title, icon);
+
+    m_text_dirty = false;
+    m_title_dirty = false;
+    m_icon_dirty = false;
+
+    m_id = response->id();
+    m_showing = true;
+}
+
+bool Notification::update()
+{
+    VERIFY(m_id != -1 && m_showing);
+
+    bool is_checked = false;
+
+    auto connection = NotificationServerConnection::construct();
```

Comment:
So you create a new connection for every single update? How does the NotificationServer check whether updates come from the same sender? If it doesn't, then other (malfunctioning/malicious) programs might interfere with the notification. That doesn't sound like a good idea.

Reply:
That was the idea behind the 63-bit notification ID's. I toyed with the idea of a continuous connection, but decided to try to keep as much code the same as possible. If you think it'd be better, we can go to a continuous connection model.

Reply:
I'm not sure how to fix this.
- One might think about keeping the connection open, and converting `s_windows` to a member of each connection, somehow, dunno whether that's possible. The client would need to keep a connection to the NotificationServer open, which sounds like a good idea anyway.
- Another way is to have some form of authentication: In order to modify a notification, the client needs to send some kind of "secret" that proves they're still the same.
- Or you say that the ID is randomized for precisely that purpose. After all, guessing a "`u63`" correctly is pretty unlikely to happen accidentally.
- Or something else entirely?

Reply:
Oh, sorry, I didn't see your reply! :D

I think the continuous connection is the way to go in the long run, but the more I think about it the more effort this seems to be. So I guess I'm fine with the ID-as-nonce idea. Could you add a TODO in the `.ipc` about that? After all, abusing IDs to also do verification seems documentation-worthy, and maybe this is a good place. What do you think?

Reply:
I'll just rework this to use a continuous model I think. I played around with it and it removes a lot of smells, is easy to implement, and also lowers the number of required function calls. I'll plan to go this way.

Reply:
The biggest issue that this has is that ClientConnection.h is autogenerated, and I cannot store a window in that as that variable would be removed upon regeneration. I will have to keep the static hashmap.

Reply:
That sounds good.

Alternatively, in theory, if you really wanna do it: You could also extend the IPC description format, to allow injecting a member somehow. No idea whether that's a good idea or how that would look :D